src/lib/components/FornaContainer.react.js

```diff
@@ -0,0 +1,274 @@
+import * as R from 'ramda';
+
+/**
+ * This is a FornaContainer component.
+ */
+export default class FornaContainer extends Component {
+    constructor(props) {
+        super(props);
+        this.renderNewSequences = this.renderNewSequences.bind(this);
+    }
+
+    componentDidMount() {
+        const {id, height, width} = this.props;
+
+        this._fornaContainer = new PreFornaContainer('#' + id, {
+            initialSize: [width, height],
+        });
+        this.renderNewSequences();
+    }
+
+    renderNewSequences() {
+        const {sequences} = this.props;
+
+        if (this._fornaContainer) {
+            this._fornaContainer.clearNodes();
+
+            sequences.forEach(seq => {
+                const unpackedOptions = Object.assign({}, seq.options, {
+                    sequence: seq.sequence,
+                    structure: seq.structure,
+                });
+                this._fornaContainer.addRNA(seq.structure, unpackedOptions);
+            });
+        }
+    }
+
+    shouldComponentUpdate(nextProps) {
+        const {sequences, colorScheme, nodeFillColor} = this.props;
+
+        if (Boolean(sequences) !== Boolean(nextProps.sequences)) {
+            return true;
+        }
+
+        if (sequences.length !== nextProps.sequences.length) {
+            return true;
+        }
+
+        const sequence_compare = R.zip(sequences, nextProps.sequences);
+
+        if (
+            sequence_compare.some(
+                seq =>
+                    seq[0].sequence !== seq[1].sequence ||
+                    seq[0].structure !== seq[1].structure
+            )
+        ) {
+            return true;
+        }
+
+        if (
+            sequence_compare.some(
+                seq =>
+                    Boolean(seq[0].options) !== Boolean(seq[1].options) ||
+                    (Boolean(seq[0].options) &&
+                        Boolean(seq[1].options) &&
+                        R.symmetricDifference(
+                            Object.keys(seq[0].options),
+                            Object.keys(seq[1].options)
+                        ).length > 0)
+            )
+        ) {
+            return true;
+        }
+
+        if (
+            sequence_compare.some(
+                seq =>
+                    Boolean(seq[0].options) &&
+                    Boolean(seq[1].options) &&
+                    Object.keys(seq[0].options).some(
+                        _option =>
+                            seq[0].options._option !== seq[1].options._option
+                    )
+            )
+        ) {
+            return true;
+        }
+
+        this._fornaContainer.addCustomColors(nextProps.customColors);
+
+        if (nodeFillColor !== nextProps.nodeFillColor) {
+            this._fornaContainer.setOutlineColor(nextProps.nodeFillColor);
+        }
+
+        this._fornaContainer.changeColorScheme(colorScheme);
+
+        return false;
+    }
+
+    render() {
+        this.renderNewSequences();
```

Comment:
Seems a little weird to put this in `render` - so it'll do nothing on initial render, because `if (this._fornaContainer)` will fail, but then on updates it happens *first*, then we depend on react keeping the same element in place as it rerenders.

Might be more natural to put this in `componentDidUpdate`, which would be more analogous to `componentDidMount` where the sequence rendering is handled on first render.